runbot_pylint/models/runbot_build.py

```diff
@@ -0,0 +1,217 @@
+# -*- encoding: utf-8 -*-
+##############################################################
+#    Module Writen For Odoo, Open Source Management Solution
+#
+#    Copyright (c) 2011 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
+#    info Vauxoo (info@vauxoo.com)
+#    coded by: moylop260@vauxoo.com
+############################################################################
+
+"""
+This module is used to create new fields in the inherited classes
+to work with pylint from runbot
+"""
+
+from openerp import models, api, fields
+import logging
+import os
+import stat
+from openerp.tools.safe_eval import safe_eval
+
+
+_logger = logging.getLogger(__name__)
+
+
+def get_depends(modules, addons_paths, depends=None):
+    """
+    Get recursive depends from addons_paths and modules list
+    """
+    if depends is None:
+        depends = []
+    for module in (modules or '').split(','):
+        if module not in depends:
+            depends.append(module)
+            for addons_path in (addons_paths or '').split(','):
+                addons_path = addons_path.strip()
+                fname_openerp = os.path.join(addons_path, module,
+                                             '__openerp__.py')
+                if os.path.isfile(fname_openerp):
+                    module_depends_list = safe_eval(
+                        open(fname_openerp, "r").read()).get('depends', [])
+                    if module_depends_list:
+                        module_depends_str = ','.join(module_depends_list)
+                        get_depends(module_depends_str, addons_paths,
+                                    depends=depends)
```

Comment:
It's strange to have a recursive method and not store the result after the call.